analyses/long-format-table-utils/annotator/annotator-api.R

```diff
@@ -0,0 +1,358 @@
+      if (!(col_add %in% available_ann_columns)) {
+        stop(paste0(
+          col_add, " is not available.\n",
+          "Available annotation columns are ",
+          char_vec_to_str(available_ann_columns), "."))
+      }
+      if (col_add %in% colnames(long_format_table)) {
+        stop(paste0(col_add, " is already in the input long_format_table."))
+      }
+    }
+  )
+  # Keep the order of columns for the returned table
+  ret_tbl_col_order <- c(colnames(long_format_table), columns_to_add)
+
+  # Detect the ".git" folder -- this will in the project root directory. Use
+  # this as the root directory to ensure proper execution, no matter where it is
+  # called from.
+  #
+  # This only works if the working directory is OpenPedCan-analysis or a
+  # subdirectory of OpenPedCan-analysis
+  #
+  # root_dir is the absolute path of OpenPedCan-analysis
+  #
+  # Adapted from the oncoprint-landscape module.
+  tryCatch(
+    {
+      root_dir <- rprojroot::find_root(rprojroot::has_file(".git/index"))
+    },
+    error = function(err_cond) {
+      # adapted from http://adv-r.had.co.nz/Exceptions-Debugging.html
+      err_cond$message <- paste0(
+        err_cond$message,
+        "\nTry re-running this function with working directory as ",
+        "OpenPedCan-analysis or a subdirectory of OpenPedCan-analysis.\n")
+      stop(err_cond)
+    }
+  )
+
+  # Read annotation data and remove duplicated rows
+  # Gene ENSG ID -> gene full name and protein refseq ID
+  ensg_gname_prt_refseq_df <- dplyr::distinct(readr::read_tsv(
+    file.path(root_dir, "analyses", "long-format-table-utils", "annotator",
+              "annotation-data", "ensg-gene-full-name-refseq-protein.tsv"),
+    col_types = readr::cols(.default = readr::col_guess()),
+    guess_max = 100000))
+  # Gene hugo symbol -> OncoKB cancer gene and oncogene/TSG
+  hgsb_oncokb_cgene_oncogene_tsg_df <- dplyr::distinct(readr::read_tsv(
+    file.path(root_dir, "analyses", "long-format-table-utils", "annotator",
+              "annotation-data", "oncokb-cancer-gene-list.tsv"),
+    col_types = readr::cols(.default = readr::col_guess()),
+    guess_max = 100000))
+  # Gene hugo symbol -> gene type
+  hgsb_gtype_df <- dplyr::distinct(readr::read_tsv(
+    file.path(root_dir, "analyses", "fusion_filtering", "references",
+              "genelistreference.txt"),
+    col_types = readr::cols(.default = readr::col_guess()),
+    guess_max = 100000))
+  # Gene ENSG ID -> RMTL
+  ensg_rmtl_df <- dplyr::distinct(readr::read_tsv(
+    file.path(root_dir, "data", "ensg-hugo-rmtl-v1-mapping.tsv"),
+    col_types = readr::cols(.default = readr::col_guess()),
+    guess_max = 100000))
+  # cancer_group -> EFO and MONDO
+  cgroup_efo_mondo_df <- dplyr::distinct(readr::read_tsv(
+    file.path(root_dir, "data", "efo-mondo-map.tsv"),
+    col_types = readr::cols(.default = readr::col_guess()),
+    guess_max = 100000))
+
+
+  # Check no NA or duplicate in the key columns of annotation data tables
```

Comment:
There is some code duplication here as multiple tables are checked for NAs and duplication. Would it make sense to create NA- and duplication-checking functions?

Reply:
I think creating functions won't help much, as the checkings are simple if statements.

The code duplication could be reduced by refactoring the whole R file by creating a list of all annotation tables, which may be worth doing. I will look into it.